tmk_core/common/chibios/eeprom_stm32.h

```diff
@@ -69,16 +69,23 @@
 #    endif
 #endif
 
-// DONT CHANGE
-// Choose location for the first EEPROM Page address on the top of flash
+/* Start of the emulated eeprom flash area */
 #define FEE_PAGE_BASE_ADDRESS ((uint32_t)(0x8000000 + FEE_MCU_FLASH_SIZE * 1024 - FEE_DENSITY_PAGES * FEE_PAGE_SIZE))
-#define FEE_DENSITY_BYTES ((FEE_PAGE_SIZE / 2) * FEE_DENSITY_PAGES - 1)
+/* End of the emulated eeprom flash area */
 #define FEE_LAST_PAGE_ADDRESS (FEE_PAGE_BASE_ADDRESS + (FEE_PAGE_SIZE * FEE_DENSITY_PAGES))
+/* Size of emulated eeprom */
+#define FEE_DENSITY_BYTES 1024
+/* Flash word value after erase */
 #define FEE_EMPTY_WORD ((uint16_t)0xFFFF)
-#define FEE_ADDR_OFFSET(Address) (Address * 2)  // 1Byte per Word will be saved to preserve Flash
 
-// Use this function to initialize the functionality
-uint16_t EEPROM_Init(void);
+_Static_assert(FEE_DENSITY_PAGES * FEE_PAGE_SIZE >= FEE_DENSITY_BYTES * 8,
+    "flash memory for emulated eeprom is too small; for correct functionality ensure it is at least 8x FEE_DENSITY_BYTES");
+
+#if defined(DYNAMIC_KEYMAP_EEPROM_MAX_ADDR) && (DYNAMIC_KEYMAP_EEPROM_MAX_ADDR >= FEE_DENSITY_BYTES)
+#error emulated eeprom: DYNAMIC_KEYMAP_EEPROM_MAX_ADDR is greater than the FEE_DENSITY_BYTES available
+#endif
+
+void     EEPROM_Init(void);
 void     EEPROM_Erase(void);
-uint16_t EEPROM_WriteDataByte(uint16_t Address, uint8_t DataByte);
+void     EEPROM_WriteDataByte(uint16_t Address, uint8_t DataByte);
```

Comment:
Why are we no longer returning FLASH_Status, here?  It may be useful to know if a write failed, and why

Reply:
the return code was never used, plus it is unclear what should be done in an event of an error